### PR TITLE
A4A: pressable referrals

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/1327

## Proposed Changes
This PR enables referrals for Pressable products also adding a feature flag to ensure we roll this out when all the backend code is ready.

In order to keep things simple, I've moved the `referral mode` into a separate file in order to avoid multiple conditions inside one class. I believe this will keep the code readable.

Also, new feature flag `a4a-pressable-referrals` added.

https://github.com/user-attachments/assets/b6dc1919-91bd-4fd8-8b1b-3b50145a3eb1

## Testing Instructions

- Checkout this branch to your local machine and spin it up
- Navigate to http://agencies.localhost:3000/marketplace/hosting/pressable. Switch to `referral` mode and check the behavior.
- Additionally, append `?flags=-a4a-pressable-referrals` to the url to check that without feature flag, user will see previous screen with `coming soon` notice.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/referral-mode/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+						plans={ pressablePlans }
+						onSelectPlan={ onSelectPlan }
+					/>
+				</>
+			) }
+
+			<PlanSelectionDetails
+				selectedPlan={ selectedPlan }
+				onSelectPlan={ onPlanAddToCart }
+				isLoading={ false }
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/referral-mode/index.tsx]: Nitpicking: No need to add this prop. isLoading is optional and will default to false if not provided.

File: config/a8c-for-agencies-horizon.json [hunk 2]
```diff
@@ -36,7 +36,8 @@
 		"a4a-hosting-page-redesign": true,
 		"a4a-dev-sites": true,
 		"a4a-migrations-page-v2": true,
-		"a8c-for-agencies-agency-tier": true
+		"a8c-for-agencies-agency-tier": true,
+		"a4a-pressable-referrals": false
```
Comment (jkguidaven) [inline config/a8c-for-agencies-horizon.json]: Nitpicking: I believe setting this `true` in Horizon is safe.

**Review by jkguidaven — APPROVED** [review]
Works great! I left non-blocking comments but this is good to go 🚢

File: config/a8c-for-agencies-horizon.json (hunk 2 above)
Comment (cleacos) [reply config/a8c-for-agencies-horizon.json]: Yep, allowing it at Horizon is safe

**Comment by cleacos** [comment]
> In order to keep things simple, I've moved the referral mode into a separate file in order to avoid multiple conditions inside one class. I believe this will keep the code readable.

Thank you!

**Comment by cleacos** [comment]
It works well. LGTM!

It's not related to this PR, but it seems that we are not rendering the plan name:

image

It's a simple fix, so we could fix it in this PR.

A minor style issue: the margin/padding-top here seems big:

image

**Review by cleacos — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16955308

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @andrii-lysenko, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Choose plan to refer`

Thank you in advance!

**Comment by rcanepa** [comment]
Hiding the plan's name is an intentional decision.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16955308).

**Pull request merged**